pipelines/Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+import static groovy.json.JsonOutput.toJson
+import static groovy.json.JsonOutput.prettyPrint
+
+@NonCPS
+def static parseEvictableHeroArtifacts(String fetchResponse, Integer max) {
+    def dateFmt = "yyyy-MM-dd'T'HH:mm:ss.SSS'+'"
+    def slurper = new groovy.json.JsonSlurper()
+
+    def artifacts = slurper.parseText(fetchResponse).items;
+    def records = artifacts.collect { [
+        repo: it.name,
+        pull: it.version,
+        lastModified: Date.parse(dateFmt, it.assets?.find().lastModified),
+        downloadUrl: it.assets?.find().downloadUrl
+    ] }
+
+    records.sort{ it.lastModified }
+    def recordsCount = records.size()
+    if (recordsCount <= max) {
+        return []
+    }
+    // Note: Groovy subscript operator is INCLUSIVE on both sides, so offset by 1
+    def lastToEvict = recordsCount - max - 1
+    return records[0..lastToEvict]
+}
+
+pipeline {
+    agent any
+    parameters {
+        booleanParam(name: 'doUpload', defaultValue: false, description: 'Flag controlling execution of upload.')
+        booleanParam(name: 'doEvict', defaultValue: true, description: 'Flag controlling execution of eviction.')
+        // Given a hero zip size of ~1.5 GB, a threshold of '20' should keep us at ~30 GB of storage cost
+        string(name: 'maxEntries', defaultValue: '20', description: 'Maximum number of cached artifacts.')
+    }
+    stages {
+        stage('Upload_Hero_Artifact') {
```

Comment:
have we considered using `mvn deploy` to deploy the ddf zip artifact to nexus?

Reply:
What are the advantages of that? Only one I can think of is getting checksums uploaded for free. I just didn't think it was critical since we're keeping this behind the VPN. Also, on my local test environment, I'm testing out of the context of a build and didn't want to assume maven was accessible (even though it likely would be). 

I used `curl` to keep this as simple as possible. I think keeping it simple is the right route, but if there are other advantages I'm missing, I'm open to it.

Reply:
deploy is less lines of code (`mvn deploy -pl distribution/ddf`) and handles credentials automatically (no hardcoding)

Reply:
it also handles making the artifact names unique